subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/CommandLineJavaCompiler.java

```diff
@@ -44,8 +46,7 @@ public CommandLineJavaCompiler(ExecHandleFactory execHandleFactory) {
 
     @Override
     public WorkResult execute(JavaCompileSpec spec) {
```

Comment:
This is a good suggestion, but let's not do it in this PR. 

It can be relatively straight-forward for the command line compiler, but may become a bit more involved with the daemon compiler given there is a hierarchy that sprouts into Groovy and Scala.

Reply:
Yeah, let's not do this here.